src/Centreon/Domain/PlatformTopology/PlatformTopology.php

```diff
@@ -0,0 +1,234 @@
+    private const SERVER_TYPE_MAP = 3;
+    private const SERVER_TYPE_MBI = 4;
+
+    /**
+     * Used to dynamically concatenate the thrown error when checking IP validity
+     */
+    private const KIND_SERVER = 'platform';
+    private const KIND_PARENT = 'parent platform';
+
+    /**
+     * @var string Server name
+     */
+    private $serverName;
+    /**
+     * @var int Server type
+     */
+    private $serverType;
+    /**
+     * @var string Server IP address
+     */
+    private $serverAddress;
+    /**
+     * @var string Server parent IP
+     */
+    private $serverParentAddress;
+    /**
+     * @var int Server parent id
+     */
+    private $serverParentId;
+
+    /**
+     * @return int
+     */
+    public function getServerType(): int
+    {
+        return $this->serverType;
+    }
+
+    /**
+     * @param int $serverType server type
+     *      SERVER_TYPE_CENTRAL = 0
+     *      SERVER_TYPE_POLLER  = 1
+     *      SERVER_TYPE_REMOTE  = 2
+     *      SERVER_TYPE_MAP     = 3
+     *      SERVER_TYPE_MBI     = 4
+     *
+     * @return $this
+     * @throws PlatformTopologyException
+     */
+    public function setServerType(int $serverType): self
+    {
+        // The API should not be used to add a Central to another Central
+        if (self::SERVER_TYPE_CENTRAL === $serverType) {
+            throw new PlatformTopologyException(
+                sprintf(
+                    _("You cannot link the Central '%s'@'%s' to another Central"),
+                    $this->getServerName(),
+                    $this->getServerAddress()
+                ),
+                Response::HTTP_BAD_REQUEST
+            );
+        }
+
+        // Check if the server_type is available
+        $availableServerType = [
+            self::SERVER_TYPE_POLLER,
+            self::SERVER_TYPE_REMOTE,
+            self::SERVER_TYPE_MAP,
+            self::SERVER_TYPE_MBI
+        ];
+        if (!in_array($serverType, $availableServerType)) {
+            throw new PlatformTopologyException(
+                sprintf(
+                    _("The type of platform '%s'@'%s' is not consistent"),
+                    $this->getServerName(),
+                    $this->getServerAddress()
+                ),
+                Response::HTTP_BAD_REQUEST
+            );
+        }
+        $this->serverType = $serverType;
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getServerName(): string
+    {
+        return $this->serverName;
+    }
+
+    /**
+     * @param string $serverName
+     * @return $this
+     * @throws PlatformTopologyException
+     */
+    public function setServerName(string $serverName): self
+    {
+        $serverName = filter_var($serverName, FILTER_SANITIZE_STRING);
+        if (empty($serverName)) {
+            throw new PlatformTopologyException(
+                _('The name of the platform is not consistent'),
+                Response::HTTP_BAD_REQUEST
```

Comment:
What's HTTP doing here?

Reply:
I use the constant in the error code do send the status of the query

Reply:
![image](https://user-images.githubusercontent.com/34628915/89059990-79af2a80-d362-11ea-9ae6-30a9cb3ace12.png)

Reply:
In fact, this kind of exception should be in the controller